lambda/rotator-snowflake/rotate/rotate.go

```diff
@@ -0,0 +1,169 @@
+package rotate
+
+import (
+	"bytes"
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"database/sql"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/chanzuckerberg/go-misc/keypair"
+	"github.com/chanzuckerberg/go-misc/lambda/rotator-snowflake/setup"
+	"github.com/chanzuckerberg/go-misc/snowflake"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"github.com/xinsnake/databricks-sdk-golang/aws"
+	"github.com/xinsnake/databricks-sdk-golang/aws/models"
+)
+
+func getUsers() ([]string, error) {
+	usersList := os.Getenv("CURRENT_USERS") //Comma-delimited users list
+	userSlice := strings.Split(usersList, ",")
+	return userSlice, nil
+}
+
+func buildSnowflakeSecrets(connection *sql.DB, username string, privateKey *bytes.Buffer) (map[string]string, error) {
+	userQuery := fmt.Sprintf(`SHOW USERS LIKE '%s'`, username)
+
+	connectionRow := snowflake.QueryRow(connection, userQuery)
+
+	snowflakeUser, err := snowflake.ScanUser(connectionRow)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to create snowflake user from userQuery %s", userQuery)
+	}
+	defaultRole := snowflakeUser.DefaultRole.String
+	if defaultRole == "" {
+		defaultRole = "PUBLIC"
+	}
+	privateKeyStr := privateKey.String()
+	stripHeaders := strings.ReplaceAll(privateKeyStr, "-----BEGIN RSA PRIVATE KEY-----\n", "")
+	stripFooters := strings.ReplaceAll(stripHeaders, "\n-----END RSA PRIVATE KEY-----", "")
+	keyNoWhitespace := strings.TrimSpace(stripFooters)
+	userSecrets := map[string]string{
+		"snowflake.user":            username,
+		"snowflake.role":            snowflakeUser.DefaultRole.String,
+		"snowflake.pem_private_key": keyNoWhitespace,
+	}
+
+	return userSecrets, nil
+}
+
+func updateDatabricks(currentScope string, secrets map[string]string, databricks *aws.DBClient) error {
```

Comment:
you might want to do the trick where, instead of expecting an aws.DBClient, you expect an interface. That way you can mock it and write tests without actually issuing DB requests

Reply:
I think I got it to work with an aws.DBClient interface, but not with an aws.DBClient.Secrets() interface. Most of the API calls are with secrets though--I might have to ask for help with this.

Reply:
Nevermind! Figured something out